### ISSUE TITLE
Client crashing on MP4 playback 

### ISSUE THREAD

**Issue by JOpsDev** [mastodon/mastodon-android#705]
When trying to play the media in https://mastodontech.de/@rdmueller/111158608925680528 my client crashes every single time.

**Comment by grishka** [comment]
Even in the last version?

**Comment by grishka** [comment]
The video file itself is also nothing special, it's H264 + AAC:
```
$ mediainfo /Users/grishka/Downloads/13f3c751137f6ad9.mp4 
General
Complete name                            : /Users/grishka/Downloads/13f3c751137f6ad9.mp4
Format                                   : MPEG-4
Format profile                           : Base Media
Codec ID                                 : isom (isom/iso2/avc1/mp41)
File size                                : 26.3 MiB
Duration                                 : 10 s 902 ms
Overall bit rate                         : 20.2 Mb/s
Writing application                      : Lavf60.4.100

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High@L5
Format settings                          : CABAC / 1 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 1 frame
Format settings, GOP                     : M=1, N=30
Codec ID                                 : avc1
Codec ID/Info                            : Advanced Video Coding
Duration                                 : 10 s 899 ms
Bit rate                                 : 20.0 Mb/s
Width                                    : 2 400 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 2.222
Frame rate mode                          : Variable
Frame rate                               : 30.000 FPS
Minimum frame rate                       : 29.791 FPS
Maximum frame rate                       : 30.232 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.257
Stream size                              : 26.0 MiB (99%)
Title                                    : VideoHandle
Language                                 : English
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
Codec configuration box                  : avcC

Audio
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 10 s 902 ms
Bit rate mode                            : Constant
Bit rate                                 : 247 kb/s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 48.0 kHz
Frame rate                               : 46.875 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 329 KiB (1%)
Title                                    : SoundHandle
Language                                 : English
Default                                  : Yes
Alternate group                          : 1
```

**Comment by JOpsDev** [comment]
Android Client Version is 2.1.4 (70)

**Comment by grishka** [comment]
Yes, please update to 2.1.5. I could never reproduce this crash myself but I made another attempt at fixing it.

**Comment by JOpsDev** [comment]
Just updated. Same result with 2.1.5 (71) should I try to flush the media cache? Culprit may be gone then.

**Comment by grishka** [comment]
No, videos don't go through the cache anyway, it's for images only.

Can you share a crash log?

**Comment by JOpsDev** [comment]
I just submitted a crash log thorough the Android requester. I could dust off some adb stuff and try to get this device connected if that is helpful. Would need some instructions what steps to follow then.

**Comment by grishka** [comment]
If you have adb set up, all you need is
```
adb logcat | grep AndroidRuntime
```
Also what is your device model and system version?

**Comment by JOpsDev** [comment]
I am seeing two different (but similar) error types:
```
10-01 12:00:36.072  3929  3929 E AndroidRuntime: FATAL EXCEPTION: main
10-01 12:00:36.072  3929  3929 E AndroidRuntime: Process: org.joinmastodon.android, PID: 3929
10-01 12:00:36.072  3929  3929 E AndroidRuntime: java.lang.IllegalArgumentException: Cannot set 'scaleX' to Infinity, the value must be <= 3.4028235E38
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.View.sanitizeFloatPropertyValue(View.java:18251)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.View.sanitizeFloatPropertyValue(View.java:18233)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.View.setScaleX(View.java:17598)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at g1.a0.l(SourceFile:56)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at g1.u$c.onPreDraw(SourceFile:55)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:1182)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3654)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2371)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9297)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1231)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1239)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:899)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:832)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1214)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:942)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7918)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
10-01 12:00:36.072  3929  3929 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

and also

```
10-01 11:48:54.589 29812 29812 E AndroidRuntime: FATAL EXCEPTION: main
10-01 11:48:54.589 29812 29812 E AndroidRuntime: Process: org.joinmastodon.android, PID: 29812
10-01 11:48:54.589 29812 29812 E AndroidRuntime: java.lang.IllegalArgumentException: Cannot set 'scaleX' to Infinity, the value must be <= 3.4028235E38
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.View.sanitizeFloatPropertyValue(View.java:18251)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.View.sanitizeFloatPropertyValue(View.java:18233)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.View.setScaleX(View.java:17598)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at g1.a0.x(SourceFile:119)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at g1.a0.onLayout(SourceFile:80)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.View.layout(View.java:23798)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6413)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$o.x0(SourceFile:33)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.recyclerview.widget.m.l2(SourceFile:171)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.recyclerview.widget.m.R1(SourceFile:39)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.recyclerview.widget.m.V0(SourceFile:357)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.I(SourceFile:67)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.G(SourceFile:72)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.onLayout(SourceFile:1)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.View.layout(View.java:23798)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6413)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at androidx.viewpager2.widget.ViewPager2.onLayout(SourceFile:71)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.View.layout(View.java:23798)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6413)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.View.layout(View.java:23798)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6413)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:4118)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3459)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2371)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9297)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1231)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1239)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:899)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:832)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1214)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:942)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7918)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
10-01 11:48:54.589 29812 29812 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

**Comment by grishka** [comment]
Ok, I see the problem now. I figured out that you're on ijug.social, so I found that post there, the way you'd see it. Then I did an API request: https://ijug.social/api/v1/statuses/111158610964039593
Here's the interesting part:
![image](https://github.com/mastodon/mastodon-android/assets/1478258/b874a9f4-afaa-41d2-96b0-0b66bd042eb0)
The culprit is that the video wasn't downloaded to your instance **and** its dimensions aren't known. The app then defaults to 0x0 (because, well, I assumed that *some* size will always be known) and eventually divides by zero. So in the end, there's nothing wrong with the video file itself.

**Comment by JOpsDev** [comment]
Here is another one which crashes my client: https://mstdn.social/@Jorvon_Moss/111157882446660116
```
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: OH SHI~
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: android.graphics.ImageDecoder$DecodeException: Failed to create image decoder with message 'unimplemented'Input contained an error.
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at android.graphics.ImageDecoder.nCreate(Native Method)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at android.graphics.ImageDecoder.createFromFile(ImageDecoder.java:327)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at android.graphics.ImageDecoder.-$$Nest$smcreateFromFile(Unknown Source:0)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at android.graphics.ImageDecoder$FileSource.createImageDecoder(ImageDecoder.java:585)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at android.graphics.ImageDecoder.decodeDrawableImpl(ImageDecoder.java:1827)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at android.graphics.ImageDecoder.decodeDrawable(ImageDecoder.java:1819)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at w.c.a(Unknown Source:0)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at w.j.g(SourceFile:42)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at w.j.g(SourceFile:412)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at w.j.j(SourceFile:284)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at w.j.k(Unknown Source:6)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at w.q$a.run(SourceFile:31)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at java.lang.Thread.run(Thread.java:1012)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: 	at w.m$a$a.run(SourceFile:37)
10-02 10:19:44.512  3878  3963 E AppKit_ImageCache: WTF?!
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: UrlImageLoaderRequest{desiredConfig=HARDWARE, desiredWidth=1000, desiredHeight=1000, uri=https://ijug.social/media_proxy/111160382757237085/small, memoryCacheKey='HARDWARE_1000_1000_https://ijug.social/media_proxy/111160382757237085/small', diskCacheKey='1d0589f32f648dbdd1664331224d65e5'}
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: java.lang.NullPointerException: key == null || value == null
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at android.util.LruCache.put(LruCache.java:169)
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at w.j.j(SourceFile:294)
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at w.j.k(Unknown Source:6)
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at w.q$a.run(SourceFile:31)
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at java.lang.Thread.run(Thread.java:1012)
10-02 10:19:44.513  3878  3963 W AppKit_ImageCache: 	at w.m$a$a.run(SourceFile:37)
10-02 10:19:44.513  3878  3963 W appkit-img-loader: error downloading image: UrlImageLoaderRequest{desiredConfig=HARDWARE, desiredWidth=1000, desiredHeight=1000, uri=https://ijug.social/media_proxy/111160382757237085/small, memoryCacheKey='HARDWARE_1000_1000_https://ijug.social/media_proxy/111160382757237085/small', diskCacheKey='1d0589f32f648dbdd1664331224d65e5'}
```
Seems to fit your analysis somehow :)

**Comment by grishka** [comment]
> There are additional messages about 'StrictMode policy violation' and 'android.os.DeadObjectException' if you need something from there.

No, that's just Android itself being verbose, probably related to some internal IPC stuff that broke when the app crashed

> Should we try to fix our ijug.social server as well or is such a situation not a problem besides the video not loading?

Maybe, but at least now I know exactly how to fix this on my side. I was fixing the wrong thing, I assumed MediaPlayer was returning nonsense when it's actually the API.

I should also probably handle decoding errors more gracefully in the image loader (but those don't lead to crashes anyway).

**Comment by grishka** [comment]
@JOpsDev could you please update to the newly released 2.1.6 and confirm that everything is working now?

**Comment by JOpsDev** [comment]
I can confirm 2.1.6 is not crashing on my two examples. Good work!
